### PR TITLE
Corrected German translation

### PR DESCRIPTION
**Pull request by vardai**
After translating the whole project to Hungarian, I found that the German translation had room for improvement too. This is only a pull request for the German translation, though, as I've learnt that other languages are not yet wanted by the project's creators.

Hope this time someone is around to correct the compiler's behavior on "ß".

### CODE REVIEW

**Comment by Ecco** [comment]
Hi @vardai . Thank you very much for your contribution! We currently don't have anyone fluent enough in German to either validate or reject this. Not that we don't trust your skills, but as for every other commit, we always double-check every code we merge.

If any other German speaker could chime in, that'd be awesome!

**Comment by Ecco** [comment]
Actually, we do have another awesome german contributor, @mhaupt from #595 
Maybe you guys could review each-other's contribution? That'd be awesome 😄

**Comment by vardai** [comment]
> Actually, we do have another awesome german contributor, @mhaupt from #595
> Maybe you guys could review each-other's contribution? That'd be awesome 😄

I've just incorporated @mhaupt's changes into my translation. After using my translation, we really have to start thinking about correctly incorporating umlauts into Epsilon. As mentioned in my comment 16 days ago, I've temporarily substituted them according to the German grammar rules.

**Comment by mhaupt** [comment]
Not being able to correctly display umlauts and other special characters is a pain indeed; it makes for an awkward reading experience, and an overall worse UX than would be good.

Thanks @vardai for incorporating my changes here.

I'll take care of the review now.

File: apps/code/catalog.de.i18n [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
-PythonCmathFunction = "cmath module function prefix"
-PythonColor = "Define a rgb color"
-PythonComplex = "Return a+ib"
+Python1J = "Imaginaeres i"
+PythonLF = "Zeilenvorschub"
+PythonTab = "Tabulator"
+PythonAmpersand = "Bitweise und"
+PythonSymbolExp = "Bitweise exklusiv oder"
+PythonVerticalBar = "Bitweise oder"
+PythonImag = "Imaginaerer Teil von z"
```
Comment (mhaupt) [inline apps/code/catalog.de.i18n]: This should be "Imaginaerteil von z".

File: apps/code/catalog.de.i18n [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
+PythonReal = "Echter Teil von z"
```
Comment (mhaupt) [inline apps/code/catalog.de.i18n]: This should be "Realteil von z".

File: apps/code/catalog.de.i18n [hunk 3]
```diff
@@ -1,30 +1,30 @@
...
+PythonSingleQuote = "Einfaches Anfuehrungszeichen"
+PythonAbs = "Absolute/r Wert/Groesse"
+PythonAcos = "Arkuskosinus"
+PythonAcosh = "Hyperbelkosinus"
+PythonAsin = "Arkussinus"
+PythonAsinh = "Hyperbelsinus"
+PythonAtan = "Arkustangens"
+PythonAtan2 = "Wiedergabe arctan(y/x)"
```
Comment (mhaupt) [inline apps/code/catalog.de.i18n]: This should be "Gib arctan(y/x)". The word "Wiedergabe" is more apt to describe the act of playing recorded music. As a translation for "Return", the simple way is "Gib" (imperative of "geben"). "Berechne" could also be used, but is more technical.

File: apps/code/catalog.de.i18n [hunk 4]
```diff
@@ -1,30 +1,30 @@
...
+PythonAtanh = "Hyperbeltangens"
+PythonBin = "Ganzzahl nach binaer konvertieren"
+PythonCeil = "Aufrundung"
+PythonChoice = "Zufallszahl aus der Liste"
+PythonCmathFunction = "cmath-Modul-Funktion-Praefix"
```
Comment (mhaupt) [inline apps/code/catalog.de.i18n]: Rather, "cmath-Modul-Funktionspraefix".

**Comment by vardai** [comment]
> it makes for an awkward reading experience, and an overall worse UX than would be good.

Yes. But at least now it is stringent in being awful, if I knew about the specific challenges of the current hardware, or outlook on separating the translations even further, we could as a next step properly deal with it.

Great, @mhaupt, that you are reviewing my changes now.

File: apps/code/catalog.de.i18n [hunk 5]
```diff
@@ -1,30 +1,30 @@
...
+PythonColor = "Definiere eine rgb Farbe"
```
Comment (mhaupt) [inline apps/code/catalog.de.i18n]: Rather, "Definiere eine RGB-Farbe".

File: apps/sequence/base.de.i18n [hunk 6]
```diff
@@ -1,22 +1,22 @@
...
 NoSequence = "Keine Folge"
 NoActivatedSequence = "Keine aktive Folge"
 NStart = "Startwert"
 NEnd = "Endwert"
-TermSum = "Summe der Glieder"
-SelectFirstTerm = "Erster Glied "
-SelectLastTerm = "Letztes Glied "
-ValueNotReachedBySequence = "Der Wert wird nicht von der Folge erreicht"
-NColumn = "n Spalte"
+TermSum = "Summe der Bedingungen"
```
Comment (mhaupt) [inline apps/sequence/base.de.i18n]: I am pretty sure the translation of "term" as "Glied" is correct, if this is about terms as parts of mathematical sequences ("Folgen"). So, the translations here should indeed be
"Summe der Glieder"
"Erstes Glied "
"Letztes Glied "
or, also correct,
"Summe der Terme"
"Erster Term "
"Letzter Term ".

File: apps/toolbox.de.i18n [hunk 7]
```diff
@@ -18,7 +18,7 @@ Product = "Produkt"
 ComplexAbsoluteValue = "Betrag"
 Agument = "Argument"
 RealPart = "Realer Teil"
```
Comment (mhaupt) [inline apps/toolbox.de.i18n]: I'd actually use "Realteil" and "Imaginaerteil" here.

File: apps/toolbox.de.i18n [hunk 8]
```diff
@@ -18,7 +18,7 @@ Product = "Produkt"
...
-ImaginaryPart = "Imaginarer Teil"
+ImaginaryPart = "Imaginaerer Teil"
```
Comment (mhaupt) [inline apps/toolbox.de.i18n]: See comment immediately above.

File: apps/usb/base.de.i18n [hunk 9]
```diff
@@ -1,7 +1,7 @@
-USBConnected = "DER RECHNER ANGESCHLOSSEN IST"
-ConnectedMessage1 = "Um Daten zu übertragen, verbinden"
-ConnectedMessage2 = "Sie sich von Ihrem Computer an"
-ConnectedMessage3 = "workshop.numworks.com"
-ConnectedMessage4 = "Drücken Sie die RETURN-Taste am"
-ConnectedMessage5 = "Taschenrechner oder ausstecken Sie ihn,"
+USBConnected = "DER RECHNER IST ANGESCHLOSSEN"
+ConnectedMessage1 = "Um Daten zu uebertragen, verbinden"
+ConnectedMessage2 = "Sie Sich von Ihrem Computer aus auf"
```
Comment (mhaupt) [inline apps/usb/base.de.i18n]: "... aus mit".

**Comment by mhaupt** [comment]
Thanks @vardai - I'm wondering what it takes to actually support umlauts.

**Comment by vardai** [comment]
Whow, @mhaupt, very well done! I'll go through and implement your annotations this evening.

**Comment by vardai** [comment]
After this git push party the German translation should be fine, Thanks again @mhaupt for your valuable contributions!

**Comment by mhaupt** [comment]
You're very welcome @vardai. More to come, also in terms of code. And please share - is there already an issue covering the umlaut question?

**Comment by vardai** [comment]
> You're very welcome @vardai. More to come, also in terms of code. And please share - is there already an issue covering the umlaut question?

I've just opened #673 to address the issue. Please comment, @mhaupt!

**Comment by vardai** [comment]
@Ecco We're ready for take-off!

**Comment by Ecco** [comment]
Guys, I'm speechless. Thank you so, so much! I'll comment about the umlaut on that other ticket.

**Pull request merged**